src/MongoDB/Session.c

```diff
@@ -93,6 +93,25 @@ static bool php_phongo_session_get_timestamp_parts(zval* obj, uint32_t* timestam
 	return retval;
 }
 
+static const char* php_phongo_get_transaction_state_string(mongoc_transaction_state_t state TSRMLS_DC)
+{
+	switch (state) {
+		case MONGOC_TRANSACTION_NONE:
+			return "none";
```

Comment:
> I've decided against exposing the MONGOC_TRANSACTION_ constants directly since we're trying to get rid of that for read preference.

Totally understandable.

> I've also opted against introducing constants for those strings for now, but we may consider doing that to allow users to use constants instead of having to write out the strings.

I do think there's value in introducing constants for the strings we return here. The only problem with ReadPreference was exposing the raw numeric values (technically bit flags), which have no meaning outside of libmongoc. I have no qualms about having constants for meaningful string values, as they are useful to avoid having applications hard-code magic string literals.

As for naming, we have a few options:

 * `TRANSACTION_STATE_NONE`
 * `TRANSACTION_NONE`
 * `TXN_STATE_NONE`
 * `TXN_NONE`

I can go either way with including "state". I can't imagine what else we might need Session constants for, so I'm not too worried about naming conflicts. "TXN" is a common abbreviation for "Transaction", so maybe that's a no-brainer -- and "TXN_STATE_" is still shorter than "TRANSACTION_".

Reply:
As for including "state", I normally would opt for including it to avoid future naming conflicts. However, `TXN_STARTING` (or `TRANSACTION_STARTING`) reads better than the other options, so I'd omit "state" for now.

As for `TXN` versus `TRANSACTION`, I normally opt against abbreviations but I'll defer to your decision here. I can live with either.

Reply:
If we're skipping `STATE`, let's use `TRANSACTION`. `TRANSACTION_STATE_` was really the only one that gave me pause due to its length.